llpc/tool/llpcPipelineBuilder.cpp

```diff
@@ -161,13 +166,17 @@ Result PipelineBuilder::buildGraphicsPipeline(BinaryData &outBinaryData) {
   PipelineBuildInfo localPipelineInfo = {};
   localPipelineInfo.pGraphicsInfo = pipelineInfo;
   void *pipelineDumpHandle = runPreBuildActions(localPipelineInfo);
-  auto onExit = make_scope_exit([&] { runPostBuildActions(pipelineDumpHandle, outBinaryData); });
 
   Result result = m_compiler.BuildGraphicsPipeline(pipelineInfo, pipelineOut, pipelineDumpHandle);
   if (result != Result::Success)
     return result;
 
+  SmallVector<BinaryData, 1> pipelines;
   outBinaryData = pipelineOut->pipelineBin;
+  pipelines.push_back(outBinaryData);
+
+  auto onExit = make_scope_exit([&] { runPostBuildActions(pipelineDumpHandle, pipelines); });
+
   return Result::Success;
```

Comment:
This change brakes the existing behavior. The clean up code won't be run when `BuildGraphicsPipeline` fails. And if we did want to run it just before returning, `runPostBuildActions` should be called directly.

Reply:
i think we can just move the code like this.

  PipelineBuildInfo localPipelineInfo = {};
  localPipelineInfo.pGraphicsInfo = pipelineInfo;
  void *pipelineDumpHandle = runPreBuildActions(localPipelineInfo);
  SmallVector<BinaryData, 1> pipelines;
  auto onExit = make_scope_exit([&] { runPostBuildActions(pipelineDumpHandle, pipelines); });

  Result result = m_compiler.BuildGraphicsPipeline(pipelineInfo, pipelineOut, pipelineDumpHandle);
  if (result != Result::Success)
    return result;

  pipelines.push_back(pipelineOut->pipelineBin);

Reply:
@kuhar  what do you think about jiaolu's suggestion?

Reply:
@jiaolu @JaxLinAMD This would restore the original behavior, which seems right to me. However, I think we should consider a less fragile solution that would allow you to work on multi-pipeline compilation internally, without changing the existing code around something that does not exist in the public. I made a suggestion in the 'Extends to multiple BinaryDatas' thread on our mailing list. Please reply what you think there, @jiaolu.